src/ui/controls/color-dropdown/index.tsx

```diff
@@ -0,0 +1,170 @@
+                element.toggleAttribute('contentEditable', false);
+                props.onColorChange(element.innerText);
+                setTimeout(function () { // Handy hack to fix something stupids in the renderBody function of the compiler
+                    element.innerText = '';
+                });
+            }
+        };
+    }
+
+    function createListItem(value: string) {
+        if (value.startsWith('#')) {
+            return (
+                <div style="display: inline-flex">
+                    <div class="color-div" id="color-div" style={'background-color: ' + value}/>
+                    <span
+                        class={{
+                            'color-dropdown__list__item': true,
+                            [props.class]: props.class != null,
+                        }}
+                        onclick={() => {
+                            store.isOpen = false;
+                            context.refresh();
+                            props.onChange(value);
+                        }}
+                    >
+                        {value}
+                    </span>
+                </div>
+            );
+        } else {
+            return (
+                <span
+                    class={{
+                        'color-dropdown__list__item': true,
+                        'color-dropdown__list__item--selected': value === props.selected,
+                        [props.class]: props.class != null,
+                    }}
+                    onclick={() => {
+                        store.isOpen = false;
+                        context.refresh();
+                        props.onChange(value);
+                    }}
+                >
+                    {value}
+                </span>
+            );
+        }
+    }
+
+    return (
+        <span
+            class={{
+                'color-dropdown': true,
+                'color-dropdown--open': store.isOpen,
+                [props.class]: Boolean(props.class),
+            }}
+        >
+            <span
+                class="color-dropdown__list"
+                oncreate={saveListNode}
+            >
+                {values
+                    .slice()
+                    .sort((a, b) => a === props.selected ? -1 : b === props.selected ? 1 : 0)
+                    .map(createListItem)}
+            </span>
+            <span
+                class={props.selected.startsWith('#') ? 'color-dropdown__selected__two' : 'color-dropdown__selected'}
+                oncreate={saveSelectedNode}
+                onclick={onSelectedClick}
+            >
+                {props.selected.startsWith('#') ?
+                    <div style="display: inline-flex">
+                        <div class="color-div__2" id="color-div__2" style={'background-color: ' + props.selected}/>
+                        <span
+                            class='color-dropdown__selected__text'
+                            id='custom_color'
```

Comment:
It is not good to use IDs in reusable components.

Reply:
The only way to get an element..